numpy/lib/function_base.py

```diff
@@ -115,13 +124,28 @@ def scott(x):
         Scott Estimator
         The binwidth is proportional to the standard deviation of the data and
         inversely proportional to the cube root of data size (asymptotically optimal)
-
         """
         h = 3.5 * x.std() * x.size ** (-1.0 / 3)
         if h > 0:
             return np.ceil(x.ptp() / h)
         return 1
 
+    def doane(x):
+        """
+        Doane's Estimator
+        Improved version of Sturges' formula which works better for non-normal
+        data.
+        See http://stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+        """
+        if x.size > 2:
+            stdskew = np.sqrt(6.0 * (x.size - 2) / ((x.size + 1.0) * (x.size + 3)))
+            sigma = np.std(x)
+            if sigma > 0:
+                mu = np.mean(x)
+                skew = (np.mean(x**3) - 3 * mu * sigma**2 - mu**3) / sigma**3
```

Comment:
Here is a paper with Doane listed as the author: http://www.amstat.org/publications/jse/v19n2/doane.pdf

My original attraction to the formula that I used is that by expanding the cubes it avoids creating some potentially large temporary arrays (3 x-sized arrays for the simple formula, only 1 for the one I am using). It is also easier to use for calculations done by hand, which it not relevant here. The simple formula does have the advantage of being numerically stable. If temp arrays are not a valid consideration, I would be glad to replace the formula. If you prefer, I can even add a flag selecting which method will be used.